hector_pose_prediction_ros/include/hector_pose_prediction_ros/visualization.h

```diff
@@ -0,0 +1,226 @@
+#ifndef HECTOR_POSE_PREDICTION_ROS_VISUALIZATION_H
+#define HECTOR_POSE_PREDICTION_ROS_VISUALIZATION_H
+
+#include <eigen_conversions/eigen_msg.h>
+#include <hector_math/types.h>
+#include <hector_pose_prediction_interface/types.h>
+#include <std_msgs/ColorRGBA.h>
+#include <visualization_msgs/MarkerArray.h>
+
+namespace hector_pose_prediction_ros
+{
+namespace visualization
+{
+template<typename Scalar>
+void drawSupportPolygonImpl( visualization_msgs::MarkerArray &marker_array,
```

Comment:
I would put the impl in a namespace as to not clutter the actual namespace.
Also declare it at the top but define it at the bottom since it's not public interface.

Reply:
Also not sure why this is an impl and not just an overload?
Could also have its uses but I would probably call `colors` -> `edge_colors`  and document how the colors are applied.